builder/azure/arm/config.go

```diff
@@ -721,3 +725,23 @@ func isValidAzureName(re *regexp.Regexp, rgn string) bool {
 		!strings.HasSuffix(rgn, ".") &&
 		!strings.HasSuffix(rgn, "-")
 }
+
+func (c *Config) validateLocationZoneResiliency(say func(s string)) {
+	// Docs on regions that support Availibility Zones:
+	//   https://docs.microsoft.com/en-us/azure/availability-zones/az-overview#regions-that-support-availability-zones
+	// Query technical names for locations:
+	//   az account list-locations --query '[].name' -o tsv
+
+	var zones = make(map[string]struct{})
+	zones["westeurope"] = struct{}{}
+	zones["centralus"] = struct{}{}
+	zones["eastus2"] = struct{}{}
+	zones["francecentral"] = struct{}{}
+	zones["northeurope"] = struct{}{}
+	zones["southeastasia"] = struct{}{}
+	zones["westus2"] = struct{}{}
+
```

Comment:
Azure really should have an API for this information. I've asked them what the plan is...

Reply:
I created a resource group in `ukwest` with an VM having `availibility zone 3` (whatever that may be) and created an image from that VM with `zone resiliency` enabled in that same resource group by hand. Guess what, the VM and image are located in `westeurope` while the resource group is in `ukwest` 😕 Beats me.